tests/components/wolflink/test_config_flow.py

```diff
@@ -0,0 +1,139 @@
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}, data=INPUT_CONFIG
+        )
+
+        result_create_entry = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"device_name": CONFIG[DEVICE_NAME]},
+        )
+
+    assert result_create_entry["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result_create_entry["title"] == CONFIG[DEVICE_NAME]
+    assert result_create_entry["data"] == CONFIG
+
+
+async def test_form_invalid_auth(hass):
+    """Test we handle invalid auth."""
+    with patch(
+        "homeassistant.components.wolflink.config_flow.WolfClient.fetch_system_list",
+        side_effect=InvalidAuth,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}, data=INPUT_CONFIG
+        )
+
+    assert result["type"] == "form"
+    assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle cannot connect error."""
+    with patch(
+        "homeassistant.components.wolflink.config_flow.WolfClient.fetch_system_list",
+        side_effect=ConnectError,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}, data=INPUT_CONFIG
+        )
+
+    assert result["type"] == "form"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_unknown_exception(hass):
+    """Test we handle cannot connect error."""
+    with patch(
+        "homeassistant.components.wolflink.config_flow.WolfClient.fetch_system_list",
+        side_effect=Exception,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}, data=INPUT_CONFIG
+        )
+
+    assert result["type"] == "form"
+    assert result["errors"] == {"base": "unknown"}
+
+
+async def test_already_configured_error(hass):
+    """Test already configured while creating entry."""
+    with patch(
+        "homeassistant.components.wolflink.config_flow.WolfClient.fetch_system_list",
+        return_value=[DEVICE],
+    ), patch("homeassistant.components.wolflink.async_setup_entry", return_value=True):
+
+        MockConfigEntry(
+            domain=DOMAIN, unique_id=CONFIG[DEVICE_ID], data=CONFIG
+        ).add_to_hass(hass)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}, data=INPUT_CONFIG
+        )
+
+        result_create_entry = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"device_name": CONFIG[DEVICE_NAME]},
+        )
+
+    assert result_create_entry["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result_create_entry["reason"] == "already_configured"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)